tests/actor/test_battery_status.py

```diff
@@ -353,6 +355,15 @@ async def test_sync_update_status_with_messages(
         )
         assert tracker._update_status(select) is None  # type: ignore[arg-type]
 
+        # Check if NaN capacity changes the battery status.
+        select = FakeSelect(battery=battery_data(component_id=BATTERY_ID))
```

Comment:
Maybe you can inherit from the real `Select` to avoid having to put a `type: ignore` everywhere?

Reply:
Yeah, I meant only to make `FakeSelect` inherit from `Select`, just that (not to rewrite any tests), but maybe it doesn't simply work if you inherit because some real methods are used then.